README.md

```diff
@@ -0,0 +1,314 @@
+
+After logging in with valid credentials and providing valid `redirect_URI` user is pushed back to the application and can retrieve response. Optionally, `authState` can be updated as well. For more information about `authState` visit [AppAuth documentation](https://github.com/openid/AppAuth-Android/blob/master/README.md).
+
+```java
+@Override
+protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+    super.onActivityResult(requestCode, resultCode, data);
+ if (requestCode == RC_AUTH) {
+        AuthorizationResponse resp = AuthorizationResponse.fromIntent(data);
+  AuthorizationException ex = AuthorizationException.fromIntent(data);
+
+  OPTIONALLY UPDATE AUTH STATE MANAGER
+
+  exchangeAuthorizationCode(resp);
+
+  } else {
+        System.out.println("Error, wrong response code");
+  }
+}
+```
+
+### 4. Requesting token
+
+With the response back `exchangeAuthorizationCode` can be triggered and token request performed.
+
+```java
+private void exchangeAuthorizationCode(AuthorizationResponse authorizationResponse) {
+    performTokenRequest(authorizationResponse.createTokenExchangeRequest());
+}
+private void performTokenRequest(TokenRequest request) {
+    mAuthService.performTokenRequest(
+            request,
+  (tokenResponse, ex) -> {
+                receivedTokenResponse(tokenResponse, ex);
+  mAuthStateManager.updateAfterTokenResponse(tokenResponse, ex);
+  });
+}
+```
+On response token details can be viewed and used to construct `authHeader` used during Apollo Client initialization.
+
+```java
+private void receivedTokenResponse(
+        @Nullable TokenResponse tokenResponse,
+  @Nullable AuthorizationException authException) {
+    mAuthState.update(tokenResponse, authException);
+
+  startActivity(new Intent(this, MainActivity.class));
+}
+```
+## Unifiedpush implementation
+
+#### Dependencies Required
+Add the following to your `build.gradle` file:
+
+```java
+apply plugin: 'com.google.gms.google-services'
+```
+
+```java
+implementation 'org.jboss.aerogear:aerogear-android-push:5.1.0'
+```
+
+### 1. External Setup
+For setting up push notifications a [Firebase](https://firebase.google.com/) account is required. The following information below is required and can be found the firebase project settings.
+
+- The `google-services.json` file.
+- Sender ID 
+- Server Key
+
+Create application within the [Aerogear Unifiefpush Server](https://github.com/aerogear/aerogear-unifiedpush-server) . For creating the application `Sender Id` and `Server Key` are required, both values can be obtained from Firebase.
+
+Once the application variant has been set, add to the `push-config.json` file.
+
+- Server URL
+- Variant ID
+- Variant Secret
+- Sender ID (`Sender Id` obtained from Firebase)
+
+### 2. Project Setup
+ Edit the `AndroidManifest.xml` to allow permissions for the push notifications.
+
+```xml
+    <uses-permission android:name="android.permission.GET_ACCOUNTS" />
```

Comment:
this doesn't look like the right permission to ask for push notification?

Reply:
We have looked into, this was a requirement for the older version of android. 

We will remove.